emu/src/cpu/arm7tdmi.rs

```diff
@@ -34,6 +34,76 @@ pub struct Arm7tdmi {
     decoded_thumb: Option<ThumbModeOpcode>,
 }
 
+#[derive(Copy, Clone)]
+#[allow(dead_code)]
+enum ExceptionType {
+    Reset,
+    UndefinedInstruction,
+    SoftwareInterrupt,
+    PrefetchAbort,
+    DataAbort,
+    Irq,
+    Fiq,
+}
+
+impl ExceptionType {
+    pub const fn address(&self) -> usize {
+        match *self {
+            Self::Reset => 0x0,
+            Self::UndefinedInstruction => 0x4,
+            Self::SoftwareInterrupt => 0x8,
+            Self::PrefetchAbort => 0xC,
+            Self::DataAbort => 0x10,
+            Self::Irq => 0x18,
+            Self::Fiq => 0x1C,
```

Comment:
Where those values come from?

Reply:
I took it from the ARM datasheet:

<img width="841" alt="image" src="https://github.com/RIP-Comm/clementine/assets/5256712/76683f60-0e14-439d-bf99-646a7c9442ae">